File: server/Models/VaultKeep.cs
using keepr.Interfaces;
namespace keepr.Models;

public class VaultKeepCreationDTO
{
  public int KeepId { get; set; }
  public int VaultId { get; set; }
}

public class VaultKeep : IRepoItem<int>
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public int KeepId { get; set; }
  public int VaultId { get; set; }
  public string CreatorId { get; set; }
}

public class VaultKeepKeep : Keep
{
  public int VaultKeepId { get; set; }
}

File: client/src/models/VaultKeep.js
import {Keep} from './Keep.js';

export class VaultKeep {
  constructor(data) {
    this.id = data.id;
    this.createdAt = new Date(data.createdAt);
    this.updatedAt = new Date(data.updatedAt);
    this.keepId = data.keepId;
    this.vaultId = data.vaultId;
    this.creatorId = data.creatorId;
  }
}

export class VaultKeepKeep extends Keep {
  constructor(data) {
    super(data);
    this.vaultKeepId = data.vaultKeepId;
  }
}
